File: server/php/subscribe-user-to-list.php
<?php
require_once __DIR__ . '/vendor/autoload.php';
use Courier\Client;
use Dotenv\Dotenv;

$dotenv = Dotenv::createImmutable(__DIR__ . '/..');
$dotenv->safeLoad();

$api_key = $_ENV['COURIER_API_KEY'] ?? '';
$list_id = $_ENV['COURIER_SUBSCRIBE_USER_TO_LIST_LIST_ID'] ?? '';
$user_id = $_ENV['COURIER_SUBSCRIBE_USER_TO_LIST_USER_ID'] ?? '';

$client = new Client(apiKey: $api_key);

$client->lists->subscriptions->subscribeUser($user_id, listID: $list_id);

File: server/kotlin/SubscribeUserToList.kt
import com.courier.client.CourierClient
import com.courier.client.okhttp.CourierOkHttpClient
import com.courier.models.lists.subscriptions.SubscriptionSubscribeUserParams

fun main() {
    val apiKey = EnvLoader.getEnv("COURIER_API_KEY")
    val listId = EnvLoader.getEnv("COURIER_SUBSCRIBE_USER_TO_LIST_LIST_ID")
    val userId = EnvLoader.getEnv("COURIER_SUBSCRIBE_USER_TO_LIST_USER_ID")

    val client = CourierOkHttpClient.builder()
        .apiKey(apiKey)
        .build()

    val params = SubscriptionSubscribeUserParams.builder()
        .listId(listId)
        .userId(userId)
        .body(SubscriptionSubscribeUserParams.Body.builder().build())
        .build()

    client.lists().subscriptions().subscribeUser(params)
}

